mmv1/third_party/terraform/services/resourcemanager/resource_google_service_account_test.go

```diff
@@ -299,3 +302,135 @@ resource "google_service_account" "acceptance" {
+
+	cases := []struct {
+		name      string
+		isNew     bool
+		before    map[string]interface{}
+		after     map[string]interface{}
+		result    map[string]interface{}
+		wantError bool
+	}{
+		{
+			name:      "normal",
+			isNew:     true,
+			wantError: false,
+			before:    map[string]interface{}{},
+			after: map[string]interface{}{
+				"account_id": accountId,
+				"project":    project,
+			},
+			result: map[string]interface{}{
+				"account_id": accountId,
+				"project":    project,
+				"email":      expectedEmail,
+				"member":     expectedMember,
+			},
+		},
+		{
+			name:      "no change",
+			isNew:     false,
+			wantError: false,
+			before: map[string]interface{}{
+				"account_id": accountId,
+				"email":      "dontchange",
+				"member":     "dontchange",
+				"project":    project,
+			},
+			after: map[string]interface{}{
+				"account_id": accountId,
+				"project":    project,
+			},
+			result: map[string]interface{}{
+				"account_id": accountId,
+				"project":    project,
+			},
+		},
+		{
+			name:      "recreate",
+			isNew:     true,
+			wantError: false,
+			before: map[string]interface{}{
+				"account_id": "recreate-account",
+				"email":      "recreate-email",
+				"member":     "recreate-member",
+				"project":    project,
+			},
+			after: map[string]interface{}{
+				"account_id": accountId,
+				"project":    project,
+			},
+			result: map[string]interface{}{
+				"account_id": accountId,
+				"project":    project,
+				"email":      expectedEmail,
+				"member":     expectedMember,
+			},
+		},
+		{
+			name:      "missing account_id",
+			isNew:     true,
+			wantError: false,
+			before:    map[string]interface{}{},
+			after: map[string]interface{}{
+				"account_id": "",
+				"project":    project,
+			},
+			result: map[string]interface{}{
+				"account_id": "",
+				"project":    project,
+			},
+		},
+		{
+			name:      "missing project",
+			isNew:     true,
+			wantError: false,
+			before:    map[string]interface{}{},
+			after: map[string]interface{}{
+				"account_id": accountId,
+				"project":    "",
+			},
+			result: map[string]interface{}{
+				"account_id": accountId,
+				"project":    "",
+			},
+		},
+	}
+	for _, tc := range cases {
+		tn := tc.name
+		tc.after["name"] = "whatever"
+		if tc.isNew {
+			tc.after["name"] = ""
+			tn = tc.name + " new"
+		}
+		tc.result["name"] = tc.after["name"]
+		t.Run(tn, func(t *testing.T) {
+			diff := &tpgresource.ResourceDiffMock{
+				Before: tc.before,
+				After:  tc.after,
+				Schema: tpgresourcemanager.ResourceGoogleServiceAccount().Schema,
+			}
+			err := tpgresourcemanager.ResourceServiceAccountCustomDiffFunc(diff)
+			if tc.wantError && err == nil {
+				t.Fatalf("want error, got nil")
+			}
+			if !tc.wantError && err != nil {
+				t.Fatalf("got unexpected error: %v", err)
+			}
+			if !maps.Equal(tc.result, diff.After) {
+				t.Fatalf("got unexpected change: %v expected: %v", diff.After, tc.result)
+			}
```

Comment:
use `cmp.Diff` instead, similar to https://github.com/hashicorp/terraform-provider-google/blob/04b14be1e6d721733fcd80648f93e727a36a800a/google/functions/region_from_zone_internal_test.go#L68